src/panels/config/automation/trigger/types/ha-automation-trigger-mqtt.ts

```diff
@@ -19,29 +22,32 @@ export class HaMQTTTrigger extends LitElement implements TriggerElement {
   }
 
   protected render() {
-    const { topic, payload } = this.trigger;
     return html`
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.triggers.type.mqtt.topic"
-        )}
-        name="topic"
-        .value=${topic}
-        @value-changed=${this._valueChanged}
-      ></paper-input>
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.triggers.type.mqtt.payload"
-        )}
-        name="payload"
-        .value=${payload}
+      <ha-form
+        .schema=${SCHEMA}
+        .data=${this.trigger}
+        .hass=${this.hass}
+        .computeLabel=${this._computeLabelCallback}
         @value-changed=${this._valueChanged}
-      ></paper-input>
+      ></ha-form>
     `;
   }
 
   private _valueChanged(ev: CustomEvent): void {
-    handleChangeEvent(this, ev);
+    ev.stopPropagation();
+    const newTrigger = ev.detail.value;
+    Object.keys(newTrigger).forEach((key) =>
```

Comment:
Doesnt seem to

Reply:
```ts
root.addEventListener("value-changed", (ev) => {
      ev.stopPropagation();
      const schema = (ev.target as HaFormElement).schema as HaFormSchema;
      fireEvent(this, "value-changed", {
        value: { ...this.data, [schema.name]: ev.detail.value },
      });
    });
```

Reply:
https://design.home-assistant.io/#components/ha-form -> See OctoPrint (at the very bottom). Remove application path.

If a field is optional, it will be removed from the output.